configs/example/riscv/fs_linux.py

```diff
@@ -168,11 +178,17 @@ def generateDtb(system):
 system.mem_mode = mem_mode
 system.mem_ranges = [AddrRange(start=0x80000000, size=mdesc.mem())]
 
+semihosting = None
```

Comment:
The parameter is `NULL` to make it happy.

Reply:
I still need to initialize this variable so I can pass it to the constructor further down.

I guess the alternative would be to use something like

```
workload_args = dict()
if args.semihosting:
    workload_args["semihosting"] = RiscvSemihosting(
        files_root_dir=args.semihosting_root,
        cmd_line=args.kernel,
    )
if args.bare_metal:
    system.workload = RiscvBareMetal(**workload_args)
    system.workload.bootloader = args.kernel
else:
    system.workload = RiscvLinux(**workload_args)
    system.workload.object_file = args.kernel
```

but I'm not convinced that is cleaner.

Reply:
> I still need to initialize this variable so I can pass it to the constructor further down.
> 
> I guess the alternative would be to use something like
> 
> ```
> workload_args = dict()
> if args.semihosting:
>     workload_args["semihosting"] = RiscvSemihosting(
>         files_root_dir=args.semihosting_root,
>         cmd_line=args.kernel,
>     )
> if args.bare_metal:
>     system.workload = RiscvBareMetal(**workload_args)
>     system.workload.bootloader = args.kernel
> else:
>     system.workload = RiscvLinux(**workload_args)
>     system.workload.object_file = args.kernel
> ```
> 
> but I'm not convinced that is cleaner.

LGTM, works in my side.

Reply:
> ```
> workload_args = dict()
> if args.semihosting:
>     workload_args["semihosting"] = RiscvSemihosting(
>         files_root_dir=args.semihosting_root,
>         cmd_line=args.kernel,
>     )
> if args.bare_metal:
>     system.workload = RiscvBareMetal(**workload_args)
>     system.workload.bootloader = args.kernel
> else:
>     system.workload = RiscvLinux(**workload_args)
>     system.workload.object_file = args.kernel
> ```

Cloud you apply the code?

Reply:
bump